mirar/processors/zogy/zogy.py

```diff
@@ -460,18 +469,25 @@ def _apply_to_images(
             )
 
             scorr = Image(scorr_data, header=image.header.copy())
-            scorr[LATEST_WEIGHT_SAVE_KEY] = image["SCORMASK"]
+            scorr[LATEST_WEIGHT_SAVE_KEY] = diff["SCORMASK"]
 
             self.save_fits(image=scorr, path=self.get_path(scorr_image_path))
 
             self.save_fits(
-                image=Image(data=diff_rms_data, header=image.get_header()),
+                image=Image(data=diff_rms_data, header=copy(image.get_header())),
                 path=self.get_path(diff_rms_path),
             )
 
             diff[BASE_NAME_KEY] = diff_image_path.name
             diff[NORM_PSFEX_KEY] = diff_psf_path.as_posix()
             diff[UNC_IMG_KEY] = diff_rms_path.as_posix()
-
+            diff[SCI_IMG_KEY] = image[BASE_NAME_KEY]
             diff_batch.append(diff)
         return diff_batch
+
+    def check_prerequisites(
+        self,
+    ):
+        check = np.sum([isinstance(x, ZOGYPrepare) for x in self.preceding_steps])
+        if check != 1:
```

Comment:
If ZOGYPrepare needs to run before ZOGY they should just be one processor, maybe with functions split into different scripts for shoretness. The only we reason we did not before was because we might use zogy without prepare.

Reply:
That sounds right, I'll open a github issue and merge the two processors in a separate PR